assets/src/api.ts

```diff
@@ -147,6 +150,24 @@ export const fetchShapeForRoute = (routeId: RouteId): Promise<Shape[]> =>
     defaultResult: [],
   })
 
+export const fetchDetourDirections = (
+  coordinates: ShapePoint[]
+): Promise<DetourShape | null> =>
+  checkedApiCall({
+    url: "/api/detours/directions",
+    parser: detourShapeFromData,
+    dataStruct: DetourShapeData,
+    defaultResult: null,
+    fetchArgs: {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "x-csrf-token": getCsrfToken(),
+      },
```

Comment:
**idea (non-blocking):** now that we've got a few `POST` requests in this file that need these headers, it might be good to factor them out to a helper function or something.

Reply:
I might do that as a follow-up refactor PR, but I like the change.

Reply:
https://app.asana.com/0/1148853526253420/1206332571064068/f